### ISSUE TITLE
Can't detect used variables in if-end block

### ISSUE THREAD

**Issue by CheyiLin** [mpeterv/luacheck#10]
Hi,

The following Lua file 'foo.lua':
```lua
local string_format = string.format
local io_stderr = io.stderr

local bar 
if true then
    bar = function (a, b)
        io_stderr:write(string_format("hello luacheck\n"))
    end 
end
```

will generate the result:
```
$ luacheck -av foo.lua
Checking foo.lua                                  Failure
    foo.lua:1:7: unused variable string_format
    foo.lua:2:7: unused variable io_stderr
Total: 2 warnings / 0 errors in 1 file
```

Is it an expected behavior or a bug? Thanks.

**Comment by mpeterv** [comment]
Hello! That's a bug, let me look into this...

**Comment by mpeterv** [comment]
Looks like I have fixed this on master branch, can you try bleeding-edge version (`sudo luarocks install  --server=http://rocks.moonscript.org/manifests/mpeterv luacheck scm`)?

**Comment by CheyiLin** [comment]
Tested. Now it works perfectly!
Thanks for fixing it so quickly.

**Comment by mpeterv** [comment]
Nice, thanks for the report! I'll make a bug-fix release somewhere today.
